File: backend/models/recipe.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Recipe struct {
	ID      uuid.UUID `gorm:"type:uuid;primaryKey"`
	Name    string
	Notes   string
	UserID  uuid.UUID
	StyleID uuid.UUID
	Created time.Time `gorm:"autoCreateTime"`

	User    User
	Style   Style
	Batches []Batch `gorm:"foreignKey:RecipeID"`
}

func (u *Recipe) BeforeCreate(tx *gorm.DB) (err error) {
	u.ID = uuid.New()
	return
}

File: frontend/src/models/recipe.ts
import Style from "./style";

class Recipe {
    ID: string = '';
    Name: string = '';
    Notes: string = '';
    StyleID: string = '';
    Created: Date = new Date();

    Style: Style | null = null;

    static fromJSON(json: any): Recipe {
        const recipe: Recipe = {
            ID: json.ID,
            Name: json.Name,
            Notes: json.Notes,
            StyleID: json.StyleID,
            Created: json.Created,
            Style: json.Style ? Style.fromJSON(json.Style) : null,
        };
        return recipe;
    }
    static fromJSONList(jsonList: any[]): Recipe[] {
        return jsonList.map((json) => Recipe.fromJSON(json));
    }
}

export default Recipe;